### ISSUE TITLE
deep_get returns 0 if the dict is not empty but key is not found.

### ISSUE THREAD

**Issue by Cediddi** [mbello/dict-deep#2]
I've lost so many hours because of this.
A simple example is this:
```
print(dict_deep.deep_get({"a":"b"}, "a"))  # outputs 'b'
print(dict_deep.deep_get({"a":"b"}, "c"))  # outputs None
print(dict_deep.deep_get({"a":"b"}, "c.d"))  # outputs None
print(dict_deep.deep_get({"a":"b"}, "c.d.e"))  # outputs 0
print(dict_deep.deep_get({}, "a"))  # outputs None
print(dict_deep.deep_get({}, "c"))  # outputs None
print(dict_deep.deep_get({}, "c.d"))  # outputs None
print(dict_deep.deep_get({}, "c.d.e"))  # outputs 0
```
Why? Why would it return 0 if the key is more than 2 levels deep... This bug is extremely annoying.

**Comment by Cediddi** [comment]
I've wrote myself a simple deep_get function that works with nested dictionaries. This does return a deeply nested item with default value if key is not found and a convenient TypeError if the nested value is not a dictionary. This solved my need for this library.

```
def deep_get(dct, key, default=None):
    key = key.split(".")
    current_key, next_keys = key[0], key[1:]
    if isinstance(dct, dict):
        if current_key in dct:
            value = dct[current_key]
        else:
            return default
        if next_keys:
            return deep_get(value, ".".join(next_keys), default=default)
        else:
            return value
    else:
        raise TypeError(f"Value is not a dict dct={dct} key={key}")
```

**Comment by mbello** [comment]
I am reopening to address this bug. Thanks for reporting it!

**Comment by mbello** [comment]
Bug fixed as in 4.1.2.

I've also added a unit test to catch regressions.

thanks for reporting it!
